File: content/04/code/js/08-change-second-scenario/features/step_definitions/steps.js
const { Given, When, Then, Before } = require('cucumber')
const { assertThat, is } = require('hamjest')

const { Person, Network } = require('../../src/shouty')

Before(function () {
  this.network = new Network()
})

Given('Lucy is {int} metres from Sean', function (distance) {
  this.network = new Network()
  this.lucy    = new Person(this.network)

  this.lucy.moveTo(distance)
})

Given('a person named Lucy', function () {
  this.lucy = new Person(this.network)
})

Given('a person named Sean', function () {
  this.sean = new Person(this.network)
})

When('Sean shouts {string}', function (message) {
  sean = new Person(this.network)
  sean.shout(message)
  this.messageFromSean = message
})

Then('Lucy should hear Sean\'s message', function () {
  assertThat(this.lucy.messagesHeard(), is([this.messageFromSean]))
})

File: content/04/code/ruby/06-broken-step-definition-implementation/features/step_definitions/steps.rb
require 'shouty'

Given("Lucy is {int} metres from Sean") do |distance|
  @network = Shouty::Network.new
  @lucy    = Shouty::Person.new(@network)
  @sean    = Shouty::Person.new(@network)
  
  @lucy.move_to(distance)
end

Given('a person named Lucy') do
  @lucy = Shouty::Person.new(@network)
end

Given('a person named Sean') do
  @sean = Shouty::Person.new(@network)
end

When("Sean shouts {string}") do |message|
  @sean.shout(message)
  @message_from_sean = message
end

Then("Lucy should hear Sean's message") do
  expect(@lucy.messages_heard).to eq [@message_from_sean]
end
